coalib/bearlib/abstractions/ExternalBearWrap.py

```diff
@@ -0,0 +1,218 @@
+                return (description + " " +
+                        FunctionMetadata.str_optional.format(default_value),
+                        setting_type, default_value)
+
+        @classmethod
+        def get_non_optional_params(cls):
+            """
+            Fetches the non_optional_params from ``options['settings']``
+            and also normalizes their descriptions.
+
+            :return:
+                An OrderedDict that is used to create a
+                ``FunctionMetadata`` object.
+            """
+            non_optional_params = {}
+            for setting_name, description in options['settings'].items():
+                if len(description) == 2:
+                    non_optional_params[
+                        setting_name] = cls._normalize_desc(description[0],
+                                                            description[1])
+            return OrderedDict(non_optional_params)
+
+        @classmethod
+        def get_optional_params(cls):
+            """
+            Fetches the optional_params from ``options['settings']``
+            and also normalizes their descriptions.
+
+            :return:
+                An OrderedDict that is used to create a
+                ``FunctionMetadata`` object.
+            """
+            optional_params = {}
+            for setting_name, description in options['settings'].items():
+                if len(description) == 3:
+                    optional_params[
+                        setting_name] = cls._normalize_desc(description[0],
+                                                            description[1],
+                                                            description[2])
+            return OrderedDict(optional_params)
+
+        @classmethod
+        def get_metadata(cls):
+            return FunctionMetadata(
+                'run',
+                optional_params=cls.get_optional_params(),
+                non_optional_params=cls.get_non_optional_params())
+
+        @classmethod
+        def _prepare_settings(cls, settings):
+            """
+            Adds the optional settings to the settings dict in-place.
+
+            :param settings:
+                The settings dict.
+            """
+            opt_params = dict(cls.get_optional_params())
+            for setting_name, description in opt_params.items():
+                if setting_name not in settings:
+                    settings[setting_name] = description[2]
+
+        @staticmethod
+        def get_severity(result):
+            """
+            Returns the severity corresponding to the one from the
+            parsed JSON.
+
+            :param result:
+                A result dict extracted from the parsed JSON.
+            :return:
+                A RESULT_SEVERITY value.
+            """
+            if not 'severity' in result:
+                return RESULT_SEVERITY.NORMAL
+            else:
+                return RESULT_SEVERITY.str_dict[result['severity']]
```

Comment:
I want an error to be raised if an invalid severity is specified (Ex: severity=blabla) so the first one would achieve that but the second one would just put a normal severity.

Do you think it is better with an error or without?